client/src/actions/index.js

```diff
@@ -28,3 +34,34 @@ export const fetchProfile = profileId => async dispatch => {
     console.log("fetchProfile: ", err);
   }
 };
+
+export const addUpdate = (data, callback) => async dispatch => {
+  try {
+    if (data.image_url) {
+      callback();
+      dispatch({ type: LOADING_UPDATE, payload: true });
+      const formData = new FormData();
+      formData.append("image", data.image_url);
+      const config = {
+        headers: {
+          Authorization: "Client-ID b981e83d44eafce"
+        }
+      };
+      const imgrData = await axios.post(
+        "https://api.imgur.com/3/image",
+        formData,
+        config
+      );
+      data.image_url = imgrData.data.data.link;
```

Comment:
Code above uploads the image to imgr and gets back link. Not great to have client-ID on front end but otherwise would be an upload to server, then an upload to imgr, so twice as long (plus i had trouble getting it to upload to server). data.image_url starts off as file then reassigned to be the link.